core/kazoo_data/src/kzs_attachments.erl

```diff
@@ -119,7 +120,8 @@ put_attachment(#{att_handler := {Handler, Params}}=Map
             handle_put_attachment(Map, Att, DbName, DocId, AName, Contents, Options, Props);
         {'error', _} = E -> E
     end;
-put_attachment(#{server := {App, Conn}}, DbName, DocId, AName, Contents, Options) ->
+put_attachment(#{server := {App, Conn}}, DbName, DocId, AttName, Contents, Options) ->
+    AName = kz_http_util:urlencode(AttName),
```

Comment:
@icehess not sure if this should be done here or in kazoo_couch. need to take a better look at code

Reply:
I'd say in kazoo_couch: another backend may want to encode this its own way.